src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/PropertyPages/BuildPropertyPage.VisualBasic.xaml

```diff
@@ -1,6 +1,375 @@
+        <ValueEditor.Metadata>
+          <NameValuePair Name="UseMonospaceFont" Value="True" />
+        </ValueEditor.Metadata>
+      </ValueEditor>
+    </StringProperty.ValueEditors>
+  </StringProperty>
+
+  <!-- TODO: Update HelpUrl -->
+  <EnumProperty Name="RunPostBuildEvent"
+                DisplayName="When to run the post-build event"
+                Description="Specifies under which condition the post-build event will be executed."
+                HelpUrl="https://docs.microsoft.com/visualstudio/ide/reference/build-events-dialog-box-visual-basic"
+                Category="Events">
+    <EnumProperty.DataSource>
+      <DataSource HasConfigurationCondition="False"
+                  PersistedName="RunPostBuildEvent"
+                  Persistence="ProjectFileWithInterception"
+                  SourceOfDefaultValue="AfterContext" />
+    </EnumProperty.DataSource>
+    <EnumValue Name="Always"
+               DisplayName="Always" />
+    <EnumValue Name="OnBuildSuccess"
+               DisplayName="When the build succeeds"
+               IsDefault="True" />
+    <EnumValue Name="OnOutputUpdated"
+               DisplayName="When the output is updated" />
+  </EnumProperty>
+
+  <!-- TODO: Replace HelpUrl with a fwlink -->
+  <BoolProperty Name="RemoveIntegerChecks"
+                DisplayName="Remove integer overflow checks"
+                HelpUrl="https://docs.microsoft.com/visualstudio/ide/reference/advanced-compiler-settings-dialog-box-visual-basic?view=#optimizations"
+                Category="Advanced" />
+
+  <!-- TODO: Replace HelpUrl with a fwlink -->
+  <BoolProperty Name="Optimize"
+                DisplayName="Enable optimizations"
+                HelpUrl="https://docs.microsoft.com/visualstudio/ide/reference/advanced-compiler-settings-dialog-box-visual-basic?view=#optimizations"
+                Category="Advanced" />
+
+  <!-- TODO: Replace HelpUrl with a fwlink -->
+  <!-- TODO: Validation -->
+  <StringProperty Name="BaseAddress"
+                  DisplayName="DLL base address"
+                  HelpUrl="https://docs.microsoft.com/visualstudio/ide/reference/advanced-compiler-settings-dialog-box-visual-basic?view=#optimizations"
+                  Category="Advanced" />
+
+  <!-- TODO: Replace HelpUrl with a fwlink -->
+  <EnumProperty Name="DebugType"
+                DisplayName="Debug symbols"
+                Description="Specifies the kind of debug symbols produced during build."
+                HelpUrl="https://docs.microsoft.com/visualstudio/ide/reference/advanced-compiler-settings-dialog-box-visual-basic?view=#optimizations"
+                Category="Advanced">
+    <EnumProperty.Metadata>
+      <NameValuePair Name="SearchTerms" Value="debug type" />
+    </EnumProperty.Metadata>
+    <EnumValue Name="none"
+               DisplayName="No symbols are emitted" />
+    <!--
+    Note that 'pdbonly' is the same as 'full'.
+    <EnumValue Name="pdbonly"
+               DisplayName="PDB Only" />
```

Comment:
I assume our current interceptors will also apply here, to map `pdbonly` to `full` in the UI.

Reply:
On that note, I wonder if we can filter interceptors by project capability in cases where we need different behaviour per language. I'm guessing not, seeing as we don't have any `AppliesTo` on the exports today.

Reply:
We could make them capability-aware, but that would require tagging all the interceptors with an `AppliesTo` attribute or they just stop showing up. That's fine for us since we could do it all at once, but poses a problem for other teams implementing interceptors. We would need to get them to add the attributes _before_ we made the change.

We should probably do it anyway; right now we risk pulling in interceptors for things like MAUI when we don't need/want them.

Reply:
Filed https://github.com/dotnet/project-system/issues/8170 to track this.